tailcfg/tailcfg.go

```diff
@@ -2044,6 +2045,66 @@ type MapResponse struct {
 	DefaultAutoUpdate opt.Bool `json:",omitempty"`
 }
 
+// DisplayMessage represents a health state of the node from the control plane's
+// perspective. It is deliberately similar to health.Warnable as both get
+// converted into health.UnhealthyState to be sent to the GUI.
+type DisplayMessage struct {
+	// ID uniquely identifies the health issue so that clients can modify their
+	// behaviour for particular issues (e.g. "session-expired").
+	ID string
+
+	// Title is a string that the GUI uses as title for this message. The title
+	// should be short and fit in a single line.
+	Title string
+
+	// Text is an extended string that the GUI will display to the user.
+	Text string
+
+	// Severity is the severity of the DisplayMessage, which the GUI can use to
+	// determine how to display it. Maps to health.Severity.
+	Severity DisplayMessageSeverity
+
+	// ImpactsConnectivity is whether the health problem will impact the user's
+	// ability to connect to the Internet or other nodes on the tailnet, which
+	// the GUI can use to determine how to display it.
+	ImpactsConnectivity bool
+}
+
+// Compare implements comparison for sorting for DisplayMessage. The ordering is
+// arbitrary, but we need to have an ordering so we can compare two lists of
+// DisplayMessages for equality.
+func (m DisplayMessage) Compare(o DisplayMessage) int {
+	return cmp.Or(
+		cmp.Compare(m.ID, o.ID),
+		cmp.Compare(m.Title, o.Title),
+		cmp.Compare(m.Text, o.Text),
+		cmp.Compare(m.Severity, o.Severity),
+		bools.Compare(m.ImpactsConnectivity, o.ImpactsConnectivity),
```

Comment:
all these are evaluated even if the first one returns non-0. It's probably fine, but not ideal